src/App.jsx

```diff
@@ -4,22 +4,47 @@ import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
 
+const url = `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+
+const headers = {
+  Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`,
+  'Content-Type': 'application/json',
+};
+
+
 const App = () => {
 
   const [todoList, setTodoList] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
+
+  const fetchData = async () => {
+    const options = {
+      method: "GET",
+      headers: headers,
+    };
+    try {
+      const response = await fetch(url, options);
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+      const data = await response.json();
+      const todos = data.records.map(record => {
+        const todo = {
+          id: record.id,
+          title: record.fields.title
+        };
+        return todo;
+      });
+      setTodoList(todos);
+    } catch (error) {
+      console.log(error);
+    }
+  };
```

Comment:
Although the assignment does not ask you to, despite the textbook stating it, it is good practice to wrap your functions that is been called in a useEffect with a usecallback hook.

```suggestion
const fetchData = useCallback(async () => {
    const options = {
      method: "GET",
      headers: headers,
    };
    try {
      const response = await fetch(url, options);
      if (!response.ok) {
        throw new Error(`Error: ${response.status}`);
      }
      const data = await response.json();
      const todos = data.records.map(record => {
        const todo = {
          id: record.id,
          title: record.fields.title
        };
        return todo;
      });
      setTodoList(todos);
    } catch (error) {
      console.log(error);
    }
  }, []);
```

Reply:
Done. Thank you!